docs/source/daml/intro/2_Scenario.rst

```diff
@@ -0,0 +1,146 @@
+.. Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Scenarios
+=========
+
+In this section you will test the ``Token`` model from :doc:`1_Token` using DAML's inbuilt ``scenario`` language. You'll learn about the basic features of scenarios:
+
+- Getting parties
+- Submitting transactions
+- Creating contracts
+- Testing for failure
+- Archiving contracts
+- Viewing ledger and final ledger state
+
+For this section, start a new module called ``Intro_2_Scenario`` and copy across the ``Token`` contract from :doc:`1_Token`. Alternatively, use the :download:`supplied source file <daml/Intro_2_Scenario.daml>`.
+
+.. scenario_basics:
+
+Scenario Basics
+---------------
+
+A ``Scenario`` is like a recipe for a test, in which the submission of transactions by parties, as well as some external information like party identities and ledger time can be scripted.
```

Comment:
I don't quite understand why it's important that it's a recipe for a test, rather than just a test? Here's my suggested rephrasing, including moving from passive to active voice.

```suggestion
A ``Scenario`` is like a test, where you can script different parties submitting a series of transactions, to check that your templates behave as you'd expect. You can also script some some external information like party identities, and ledger time.
```

Reply:
The `recipe for` terminology is how I decided to talk about monads. An expression of type `Scenario a` is still a pure expression and can be evaluated using pen and paper using only the DAML code in front of you. But you can't actually turn it into a test outside the context of some sort of runtime that knows what `getParty` and `submit` actually do.

Similarly with `Update`. One can think of `Update` as a transaction, but it's not really. It's a "recipe for a transaction" that you can only execute to get a transaction in the context of a ledger.